### PR TITLE
Added type hints to api_v2/client modules

### PR DESCRIPTION
**Pull request by luisg5**
### Summary

Adresses #265 

### Details and comments

Added type hints to auth.py, version.py, and account.py, within the api_v2 directory.

### CODE REVIEW

File: qiskit/providers/ibmq/api_v2/clients/auth.py [hunk 1]
```diff
@@ -13,18 +13,23 @@
...
 """Client for accessing authentication features of IBM Q Experience."""
+from typing import Dict, List
 
 from ..exceptions import AuthenticationLicenseError, RequestsApiError
 from ..rest import Api, Auth
 from ..session import RetrySession
 
 from .base import BaseClient
 
+# Type aliases for type hints.
```
Comment (luisg5) [inline qiskit/providers/ibmq/api_v2/clients/auth.py]: Some annotations were used multiples times, so I defined type aliases.

File: qiskit/providers/ibmq/api_v2/clients/account.py [hunk 2]
```diff
@@ -16,20 +16,34 @@
...
+
 
 class AccountClient(BaseClient):
     """Client for accessing an individual IBM Q Experience account.
 
     This client provides access to an individual IBM Q hub/group/project.
     """
 
-    def __init__(self, access_token, project_url, websockets_url, **request_kwargs):
+    def __init__(
```
Comment (luisg5) [inline qiskit/providers/ibmq/api_v2/clients/account.py]: As a result of type hints, the length of some functions were beyond the line length limit. I decided handle it this way, but it's up in the air if there are better alternative suggestions.

**Comment by jyu00** [comment]
I think for this module there isn't really much a need to specify the type _inside_ a dictionary, especially since many of them are just responses from the api, which could change in the future. For example, 
```
def backend_status(self, backend_name: str) -> Dict[str, Union[str, int, bool]]:```

Knowing that the values of the returned dictionary can be one of `str, int, bool`, IMO, doesn't really help the user much.

File: qiskit/providers/ibmq/api_v2/clients/auth.py (hunk 1 above)
Comment (diego-plan9) [reply qiskit/providers/ibmq/api_v2/clients/auth.py]: Can you actually remove the aliases for the time being? The mean reason is starting the effort "simple and explicit", and we can revise using aliases once we are a bit more advanced in the typing effort if needed.

File: qiskit/providers/ibmq/api_v2/clients/account.py [hunk 3]
```diff
@@ -16,20 +16,32 @@
...
+# Disabled unused-import because datetime is used only for type hints.
+from datetime import datetime  # pylint: disable=unused-import
+
 from ..rest import Api
 from ..session import RetrySession
 
 from .base import BaseClient
 from .websocket import WebsocketClient
 
+# Type aliases for type hints.
```
Comment (diego-plan9) [inline qiskit/providers/ibmq/api_v2/clients/account.py]: This is probably a leftover from a previous iteration?

File: qiskit/providers/ibmq/api_v2/clients/auth.py [hunk 4]
```diff
@@ -112,7 +117,7 @@ def user_hubs(self):
         """
         response = self.client_api.hubs()
 
-        hubs = []
+        hubs = []  # type: Hubs
```
Comment (diego-plan9) [inline qiskit/providers/ibmq/api_v2/clients/auth.py]: Let's not introduce type hints _inside_ the functions during this round - as with other comments, we can reevaluate once we have a wider picture (don't worry as well it is produces warnings, etc).

File: qiskit/providers/ibmq/api_v2/clients/account.py (hunk 3 above)
Comment (luisg5) [reply qiskit/providers/ibmq/api_v2/clients/account.py]: Yeah, this was leftover. Thanks for the catch! I'll make sure to remove it 👍🏽.

File: qiskit/providers/ibmq/api_v2/clients/auth.py (hunk 4 above)
Comment (luisg5) [reply qiskit/providers/ibmq/api_v2/clients/auth.py]: It was the warnings that made me decide to add the type hints for the variables. But, that makes sense. I'll focus on the bigger picture as we go on for now. I removed them in the most recent PR.

**Pull request merged**